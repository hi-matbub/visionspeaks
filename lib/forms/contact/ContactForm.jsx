import React from 'react';
import Form from 'react-bootstrap/Form';

const ContactForm = () => (
  <Form.Row>
    <h2 className="mt-4">Get In Touch</h2>

    <Form.Control
      className="mt-3"
      type="text"
      label="email"
      name="email"
      placeholder="jane.doe@example.com"
      required
    />

    <Form.Control
      className="my-3"
      type="text"
      as="textarea"
      rows={3}
      label="message"
      name="message"
      placeholder="Inquiring about my recent purchase..."
      required
    />

    <Form.Check
      className="my-3"
      type="checkbox"
      label="I am not a robot"
      required
    />
  </Form.Row>
);

export default ContactForm;
