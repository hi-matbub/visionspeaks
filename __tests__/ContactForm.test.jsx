import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ContactForm from '../lib/forms/contact/ContactForm';

it('Contact components renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ContactForm/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Contact component renders unchanged', () => {
  const tree = renderer.create(<ContactForm/>).toJSON();
  expect(tree).toMatchSnapshot();
});
