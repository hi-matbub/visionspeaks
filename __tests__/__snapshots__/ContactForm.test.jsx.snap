// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component renders unchanged 1`] = `
<div
  className="form-row"
>
  <h2
    className="mt-4"
  >
    Get In Touch
  </h2>
  <input
    className="mt-3 form-control"
    label="email"
    name="email"
    placeholder="jane.doe@example.com"
    required={true}
    type="text"
  />
  <textarea
    className="my-3 form-control"
    label="message"
    name="message"
    placeholder="Inquiring about my recent purchase..."
    required={true}
    rows={3}
    type="text"
  />
  <div
    className="my-3 form-check"
  >
    <input
      className="form-check-input"
      disabled={false}
      required={true}
      type="checkbox"
    />
    <label
      className="form-check-label"
      title=""
    >
      I am not a robot
    </label>
  </div>
</div>
`;
